### PR TITLE
Convergence restructure

### PR DESCRIPTION
**Pull request by andrewfullard**
### :pencil: Description

**Type:**  :roller_coaster: `infrastructure`

This PR will take convergence handling out of the simulation and into a new class.

### :vertical_traffic_light: Testing

How did you test these changes?

- [x] Testing pipeline

### CODE REVIEW

**Comment by wkerzendorf** [comment]
Make different classes for different convergence strategies. 
Track all values for convergence.

**Comment by andrewfullard** [comment]
Let's get this merged to help Deeksha work on convergence

File: tardis/simulation/convergence.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
+import numpy as np
+
+
+class ConvergenceSolver:
+    def __init__(self, strategy):
+        """_summary_
```
Comment (andrewfullard) [inline tardis/simulation/convergence.py:6]: fix docstring

File: tardis/simulation/convergence.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+        Parameters
+        ----------
+        strategy : _type_
```
Comment (andrewfullard) [inline tardis/simulation/convergence.py:10]: fix docstring

**Review by Knights-Templars — COMMENTED** [review]
Please add some tests for the new functions.

File: tardis/simulation/tests/test_convergence.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    with pytest.raises(ValueError):
+        ConvergenceSolver(strategy)
+
+
+def test_damped_converge(strategy):
+    solver = ConvergenceSolver(strategy)
+    value = np.float64(10.0)
+    estimated_value = np.float64(20.0)
+    converged_value = solver.damped_converge(value, estimated_value)
+    assert converged_value == 15.0
```
Comment (wkerzendorf) [inline tardis/simulation/tests/test_convergence.py:46]: make this `npt.assert_almost_equal` or`npt.assert_close`

File: tardis/simulation/tests/test_convergence.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+
+def test_get_convergence_status(strategy):
+    solver = ConvergenceSolver(strategy)
+    value = np.array([1.0, 2.0, 3.0], dtype=np.float64)
+    estimated_value = np.array([1.01, 2.02, 3.03], dtype=np.float64)
+    no_of_cells = np.int64(3)
+    status = solver.get_convergence_status(value, estimated_value, no_of_cells)
```
Comment (wkerzendorf) [inline tardis/simulation/tests/test_convergence.py:54]:
```suggestion
    is_converged = solver.get_convergence_status(value, estimated_value, no_of_cells)
```

**Review by wkerzendorf — DISMISSED** [review]

**Pull request merged**